sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/CommonFileDataSourceSuite.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources
+
+import org.scalatest.funsuite.AnyFunSuite
+
+import org.apache.spark.sql.{Encoders, FakeFileSystemRequiringDSOption, QueryTest, Row}
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+import org.apache.spark.sql.test.SQLTestData
+
+// The trait contains tests for all file-based data sources.
```

Comment:
Can you add some comments here about `FileBasedDataSourceSuite` vs `CommonFileDataSourceSuite` by linking properly?

For example, we can say like: `CommonFileDataSourceSuite` requires to run the same codes in all file based data sources. `FileBasedDataSourceSuite` is a more loose place where we put the tests with some variants/differences.

We should also better mention that libsvm is not included.

Reply:
I included libsvm as well.